cmd/snap/cmd_create_key.go

```diff
@@ -58,7 +55,7 @@ func (x *cmdCreateKey) Execute(args []string) error {
 	if keyName == "" {
 		keyName = "default"
 	}
-	if !validKeyName.MatchString(keyName) {
+	if !asserts.ValidAccountKeyName.MatchString(keyName) {
```

Comment:
The problem is that `snap create-key` doesn't create an assertion, so there won't be such an error until much later, and I want to avoid users wasting effort generating a key with a fixed name only to find that the name is invalid.  Other suggestions for how to lay it out would be welcome, but fundamentally I think we have to export some mechanism to check whether a given name is valid.